### ISSUE TITLE
The Middle East Water Collection (middle-east-water)

### ISSUE THREAD

**Issue by wickr** [OregonDigital/OD2-migration#1120]
**Item Count:** 7,823 items

**Item Types:** Image/Document

**Access Restrictions:** 2,393

**Complex Objects:** 0

**pid List:** https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_pids.txt

https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_aa.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ab.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ac.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ad.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ae.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_af.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ag.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ah.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ai.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_aj.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ak.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_al.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_am.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_an.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ao.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ap.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_aq.txt
https://github.com/OregonDigital/OD2-migration/blob/master/middle-east-water/middle-east-water_ar.txt

**Comment by morganem0** [comment]
Re-exported collection metadata records from Solr, and created QA sheet for starting QA on checked batches. 

To create QA sheet, did following steps: 
* processed re-exported Solr data in jupyter nb, exported csv from nb, and in OR added column with batches label, to use for selecting rows from each batch for QA
* added QA sheet to google drive here:  


note: When re-exporting, initially exported from Solr with row-limits set to 3k, then concatenated csvs. But this caused duplicate rows.  
Tried export with row limit set to ~7400 to get all records at once and this worked OK, no timeouts/failed csv.

**Comment by morganem0** [comment]
## Collection QA Summary 

These QA steps are based on: 
* original 2021 template in gh issue: "[QA template to copy/paste #5](https://github.com/OregonDigital/OD2-migration/issues/5)" 
* with re-ordered and added steps from doc: "[Instructions for Post-Migration QA steps](https://docs.google.com/document/d/1n9RFCb7g32ha46Fs1kxks3Y-w10p6XFv6EOglxLndPE/edit?tab=t.8zgdbsg63yi1#heading=h.encklv8bvt4h)" 


**QA sheet for MEW  collection:**
https://docs.google.com/spreadsheets/d/1-RIVzzlC8mzOm_Y-PpmhuVLHq2Wk3cc_Ma18Z0KiLXc/edit?gid=1621936616#gid=1621936616

  
**batches, facets reports for this collection**
- [x] merged batches PIDs into QA sheet?
- [x] ^^ merged for aa-ad only. Merge addtl batches as ready
- [x] selected 5+ PIDs per batch, per Hyrax type, and concatenated Work Showpage URLs for OD2, OD2 Data Sources, and OD1. (URLs tab/, in QA sheet) 
- [x] Facets fields are in  __ tab, in sheet:  __link__ 
- [x] "Coll_level_stats_admin_counts_totals" tab in QA sheet has unique counts, total rows/non-empty, for subset of all fields:

<google-sheets-html-origin> 
Solr Field | Count - Unique values | Total values
-- | -- | --
admin_set_sim | 1 | 7823
full_size_download_allowed_label_sim | 1 | 7820
institution_label_sim | 1 | 7823
member_of_collection_ids_ssim | 1 | 7823
primarySet_ssim | 1 | 7823
rights_statement_label_sim | 1 | 7823
visibility_ssi | 3 | 7823
  |   |  
values - visibility_ssi - from OR |   | Total values
open |   | 5428
osu |   | 2392
restricted |   | 3

- restrictions count in this ticket (above) 
Access Restrictions: 2,393


### Collection-level Post-mig QA steps 
**Overall - Full Collection:**
- [x] All works are migrated- original number of works and OD2 works match (Migration)
- [x] Confirm visibility settings are correct for all works (by count).

OD1 Counts / number of works 
<google-sheets-html-origin>
OD1 |  
-- | --
open - signed out | 5430
logged in | 7823

- [x] Confirm works are in the correct Admin set.

OD2 Admin set: 
primarySet_ssim: osu

- [x] Verify all thumbnails - (Doesn’t apply to compounds) 
    - Paged through all 79 pages of Works, at 100 Works/page - page URLs in "thumbnails-pages-concatenated" tab in QA sheet. 
    - Thumbnails look good overall - only found 4 Works with no thumbnails 
    - Noted them in this sheet: "MEW_thumbnail issues" https://docs.google.com/spreadsheets/d/1GgG2l1_J9Mu6T1b8z-gN2cF1P44TrzM4ciw9jZfEbbI/edit?gid=0#gid=0

- [x] Compare facet lists between OD1 and OD2. **a majority of the issues spotted in QA have been in this step.**



**Pre-Approve Works**
- [x] Setup: Select (minimum) 5 works per batch for QA. 
- [x] Check Hyrax types and Restrictions status for Works in this collection - at the top of the GH batch-level ticket.
- [x] Include 5 Works of each Hyrax type
  - [x] If collection has different types (Documents, Images, Video, Audio, etc.), check 5 of each type, or all of a type if there are 5 or fewer. 
  - [x] If mixed Restrictions status, also include 5 restricted items (treat as a different worktype).
- [x] Pre-Approve minimum of 5 Works. 
- [x] Check that pre-approved Works are accessible in browse everything/all
- [x] Check that Showpages for pre-approved Works are accessible in by clicking on them in  browse everything/all. 


**Collection pages:**
- [x] Collection pages are there
- [x] Collection description/blurb is there
- [x] Restrictions: Double-check restricted collections and works (VPN). 
  - [x] Counts on Collection page reflect expected count with Restrictions. 


**Collection Facets:**
- [x] Collection Level Facets 
  - [x] Collection-specific facets appear on the collection homepage (after pre-approving test set of Works)
  - [x] Controlled and uncontrolled facets appear and are populated by data from works 
- [x] Labels are fetching correctly
- [x] URIs are resolving correctly

**Search/indexing/core faceting check:**
- [x] In Search Results pages, Thumbnails are appearing for works; the collection has the correct icon.
- [x] Works and collections appear in browse everything/all when searched with corresponding metadata
- [x] Keyword searching appears in browse everything/all and keywords in title, description, full text appear with hit highlighting for works and collections
- [x] Core facets - in Search results pages, Core facets show expected values based on these Works 


### Work-level Post-mig QA steps 
**Work pages:** 
- Works checklist moved to separate comment, to be added below main Coll-level checklist/comment


### QA done: sign-off/ approvals steps 
**Bulk Approve**
- [x] Any QA issues should have been noted in the batch ticket. Once all batch tickets are complete, compile any issues from those into a comment on the collection ticket. 
- [x] Ask Migration team (i.e. Ryan) to bulk review by leaving a comment in the Github ticket(s), with @wickr.  


### Final Collection-level and Admin tasks 
**Final Collection QA**
- [ ] After all Works are approved, Compare number of assets on OD2 with total in the original collection-level GH ticket. 
- [ ] after all batches pass QA - notify Migration team (Ryan W) for any last checks. Migration team will also check for any changed status in OD1. 
- if any changed OD1 Works, update and migrate those works as needed. (make a separate ticket and link to the QA ticket). 
- [ ] When all checks are done, close GH tickets.

**Comment by morganem0** [comment]
## Errors/issues

this comment is for compiling any irregularities found during Collection-level QA, to pull them out from the main checklist. 

### Issues to check 
- [x] slight discrepancy in visibility counts
  - [x] Done - now matches. Details below. 
- [x] 4 Works with thumbnails issues 
- [x] Donor field blank in 1 OD1 Work, but has Donor in OD2: Naff, Thomas$http://id.loc.gov/authorities/names/n77003981 | df70qw443
  - [x] **ticketed** https://github.com/osulp/metadata/issues/314
- [x] Topic field labels, and counts discrepancies. (details below) 
- [x] Format - 7 Works are blank in OD2, have PDF in OD1.
  - [x] **RESOLVED**. Export from Solr now shows only 2 Works blank for `file_format_sim`. I confirmed that both of these also have no files on display in OD1. Final counts for OD2: `file_format_sim` below. 
  - [x] **ticketed** for followup - granular PDF types: https://github.com/osulp/metadata/issues/315
- [x] Data Sources panel - footnote numbers not appearing. Example record: https://oregondigital.org/concern/documents/df70p8535?locale=en#data_sources . Reminder to check for existing bug tickets for this error. 
- - [x] DONE - I searched in osulp and OD repos and did not see existing bug tickets for this, so made one: https://github.com/osulp/metadata/issues/312. (We can move it into OD repo if relevant)
- [x] Water basin labels missing 'River' - details in a separate comment in this ticket. Not sure if this is an issue, just noting it for discussion. 
  - [x] DONE - shared about this at Metadeities mtg 03/25, added details to off-cycle notes doc: [geonames/labels issue for STREAM features - details](https://docs.google.com/document/d/1ZSYVFvL9RmuOC_4C4w8_2BC82qwprbp8DXTOHtQzO74/edit?tab=t.0#heading=h.26ryvyfnnqky) 



### Likely non-issues? 
(ie, check if these are expected?)
- [x] Submission Date is different in OD2 vs. OD1. not sure if this matters. 
  - [x] **OK** - checked with Ryan W, this was a mig decision. 
- [x] “Search within” is not finding known-positive terms in OD2. Not sure if this was working in OD1? OD1 files in viewer are intermittently not loading - return to test OD1 search within when viewer loading files?  
  - [x] Done - spot-checked “Search within” behavior in OD1 vs OD2 issues and there are many discrepancies. Ryan W confirmed this is expected, variance in OCR between systems 
- [x] Noting for documentation purposes only: "Language" field displays in Work showpage only after approval


### Details for errors/issues

- - ~~A few Topics are not pulling labels - ticket for fixing after approvals?~~  
 -  - RESOLVED. Checked that these have been fixed, or noted in sheet. 
 -  - See facets-counts tab in: [MEW_QA_withURLs_SolrAll_OR_20250222](https://docs.google.com/spreadsheets/d/1-RIVzzlC8mzOm_Y-PpmhuVLHq2Wk3cc_Ma18Z0KiLXc/edit?gid=106101136#gid=106101136)
--  
- - ~~a small number of Topics have huge counts discrepancies. Check these values to ID source of differences. See facets-counts tab in mainQA sheet, filter for non-match~~
 -  -  - RESOLVED. This was just a weird glitch with those numbers in sheet - not sure how it happened but now fixed. 


**visibility discrepancy resolved**
Final  visibility_ssi counts - OD2**
(counts as of 03/27/2025)
open	5430
osu	2393

original OD1 counts: 
Restricted access - OSU only 2393
(blank)	5430


**Final file_format_sim counts - OD2**
counts as of 03/27/2025.  
pdf (PDF EXIF\, Portable Document Format)	6
pdf (PDF/A)	6519
pdf (PDF/A\, PDF EXIF)	4
pdf (Portable Document Format)	1272
pdf (Portable Document Format\, PDF EXIF)	1
tiff (TIFF EXIF)	19
(blank)	2

Blanks - Final file_format_sim OD2/OD1:
https://od1.oregondigital.org/sets/middle-east-water/oregondigital:df70rr877
https://od1.oregondigital.org/sets/middle-east-water/oregondigital:df70rr89s

https://oregondigital.org/concern/images/df70rr877
https://oregondigital.org/concern/images/df70rr89s



**Topics Counts error** (now resolved)
OD1 - topics |   | OD2 - topics |  
-- | -- | -- | --
Persian Gulf War, 19 | 9138 | Kineret (Israel) | 37
Kineret (Israel) | 37 | Persian Gulf War, 1991 | 37

**Comment by wickr** [comment]
The 4 works with no thumbnails don't have a content file in OD1, so there was nothing to migrate and make derivatives for.

The visibility counts might align better when I finish full verification of the other batches.

**Comment by morganem0** [comment]
Of the 4 works, 3 appear to have no content files. 1 appears to have a PDF: 
https://oregondigital.org/concern/documents/df70rf749?locale=en

3 with no content files:
https://oregondigital.org/concern/documents/df70rr877?locale=en
https://oregondigital.org/concern/documents/df70rr89s?locale=en
https://oregondigital.org/concern/documents/df710j616?locale=en

**Comment by wickr** [comment]
Oh you're right, somehow I had the same pid in both batch ac and ad for my error tracking. So maybe only 3 so far with no content file.
I fixed the derivatives on df70rf749 

There's a way to query solr for any works missing a 'real' thumbnail, since the default thumbnail graphic is a PNG format but the content-based ones aren't, we can search for the presence of the PNG files by the extension/ending.
```solr
member_of_collection_ids_ssim:middle-east-water AND thumbnail_path_ss:*png
```

at the moment that gives 7 results, the 4 not yet identified are:
df70rh60c, df70q6589, df70p677b, df70r809z

I'll work on those now.

**Comment by wickr** [comment]
Ok got those 4 files/derivatives fixed up.
3 of those (df70q6589, df70p677b, df70r809z) were marked as Private/restricted, so they didn't have the update done yet. I set those to OSU. 

However, only 1 (df70r809z) actually has the Access Restrictions value set, that might be the only one needing 'OSU' the other 2 are 'public' I think. I'll need to double check, but I think that would make the counts line up.

**Comment by morganem0** [comment]
### Collection-level Facets QA outcomes

Overall Collection-level facets look OK. There are a couple facet fields that will need to wait for confirming totals after completing Works-level QA and approving all Works. More details below about these. I checked about these with Ryan W; he confirmed the earlier stages in migration were simpler for Collection-level facets checking, and at this stage it’s not feasible for some cases. 

#### Facets QA - summary
In MEW, there are a couple categories of facets that need to wait for final confirmation after Works-level review.   
1. A couple facets are challenging/prohibitively complicated to check completely before Works are approved, because the facet is: 1) not available in the Review Submissions queue, **AND** 2) the field or fields for this facets contains multiple/complex combinations of values that make it challenging to match OD2 values from the Solr export, with the exported OD1 values. These are: 
Water Basin
Publisher


2. Creator-like fields. 
In OD1, there are separate facets for Creator, Contributor, Donor. In OD2, the Collection-specific PUI facets can include Contributor and Donor, but in OD2 Review Page (Admin UI), Contributor and Donor are not available as sep facets. Also in OD2, the Contributor and Donor fields are included into OD2 Creator facet. So, if visually checking the Creator facet counts, there is not a simple/direct comparison available. Checking values in the "creator-like" fields for exported data from OD1 vs OD2 is also challenging because of the multiple fields, most of which with multiple values, resulting in some cases in different combinations showing as discrete values for the field counts within Open Refine, and none of which would match to what is presented in the UI facets without time-consuming work to recreate those combinations. 
The approach I will use instead is filtering after approving Works to check facet counts for selected creator/contributor/donor combinations. 

For other facets that are not available in the Review Submissions queue, the field values in OD2 Solr were similar enough to OD1 that it was possible to check them prior to approval. 
These were: 
Donor
Repository
Type /(Media in OD2)
 
Donor field had minor discrepancies. 
2 Works in OD1 have correct URI, but did not have label displaying. These Works’ labels are correct in OD2, so this was a problem in OD1 only. 

1 OD1 work is blank in the Donor field, but has Donor in OD2. I will add to errors/issues list for followup. 
Naff, Thomas$http://id.loc.gov/authorities/names/n77003981
df70qw443

these facets counts look OK: 
Repository
Type /(Media in OD2)


Remaining facet fields were checked via comparing OD1 PUI/OD2 Review Submissions queue values, or exported metadata values in openrefine. 
Topic
Work Type
Region
Language
Format
Collection
Institution
Decade

The Format field had small discrepancies for some Works. 
These Works are viewable in spreadsheet: 
[MEW_format_issues](https://docs.google.com/spreadsheets/d/1u0-i8eX3EIHWudRf7LLBPKnD-U3tL_AG6iiQS89ufU8/edit?gid=1785424258#gid=1785424258) see “issues_PIDs” tab

**summary of Format issues:** 
Work has PDFA in OD2, but TIFF in OD1: 
df70rn55b

3 Works have TIFF in OD2, but PDF in OD1: 
df70rq693
df70rq68t
df70rq67j

7 Works are blank in OD2, have PDF in OD1.

**Comment by morganem0** [comment]
## Works-level QA 
**Done**


### Summary of Works tested
Works checked from this batch - in tab1, in sheet https://docs.google.com/spreadsheets/d/1-RIVzzlC8mzOm_Y-PpmhuVLHq2Wk3cc_Ma18Z0KiLXc/edit?gid=1621936616#gid=1621936616

232 Works total, from batches aa-ap.  Any minor issues with individual works noted in column "QA-status".  

Works selected based on: 
filter for: 
- batch (at least 5 from each batch of 500: AA, AB, AC, etc)
  - within batches, visibility_ssi  (at least 5 from each value: osu, open) 
  - water_basin_label_sim: not blank
  - language_label_sim: select all non-English 1st to include (this was not many Works total; rest will be all English [eng]) 
  - location_label_sim: choose for multiple values, and distinct values in QA set 


### Checklist Steps from template 

**Work pages:** 
complete the checks below for each of the minimum-5 test Works (check all steps at the work page)
- [x] The digital object(s) appear in viewer
- Files are attached
- Derivatives were created (both automated check from Migration and visual check from Metadeities)
- [x] Metadata is also there in work and viewer
- Collection information is also on work page
- Confirm metadata is in IIIF viewer
- [x] Linked Data:
- Labels are fetching correctly
- URIs are resolving correctly
- [x] Download
- check that file downloads, matches content in viewer in OD1
- [x] - [x] Copyright 
- Confirm copyright logo matches statement

**Comment by morganem0** [comment]
### Water Basin labels missing 'River' 

**Done**. Ticketed: https://github.com/osulp/metadata/issues/313


~~I am just noting this here for eventual followup with Ryan W. No need to address immediately!~~

In the Water basin field, many OD2 labels are missing “River”, vs the values from OD1. 

example: 
In OD1, “Jordan River”
example Work metadata: 
Water Basin: Jordan River >> Israel

In OD2, “Jordan”
example Work metadata: 
Water Basin: Jordan >> Israel

Checking geonames for a few affected URIs, I am not seeing preflabels currently - maybe preflabels were there when the OD1 works were originally created?

example Works: 
https://od1.oregondigital.org/sets/middle-east-water/oregondigital:df710k22q#page/1/mode/1up
https://oregondigital.org/concern/documents/df710k22q?locale=en

OD1
Water Basin
Jordan River >> Israel
Euphrates River >> Basra Governorate >> Iraq

OD2
Water Basin
Jordan >> Israel
Euphrates River >> Basra >> Iraq

There are not very many unique OD2 Water Basin URIs/Labels overall that don’t have River, but there are a lot of Works that are from “Jordan River”, and that label is missing 'River' - so this affects a large number of Works overall. The Water Basin labels that are missing River are also confusing when they're showing up in the Location facet, because it’s causing “Jordan >> Israel” to show up in the Location facet, along with the value for Location: “Jordan”.   

It looks like the Location facet has combined values from Water Basin and Location in: 1) the main "[View All" results for MEW,](https://oregondigital.org/catalog?locale=en&f[non_user_collections_ssim][]=middle-east-water), and in 2) the Review Submissions page.  However, in the [MEW Collection page](https://oregondigital.org/collections/middle-east-water?locale=en), the Collection facets include both Location and Water Basin  as separate facets. 

**Water Basin labels missing River - OD2 vs OD1**

OD 2 - Water Basin - Labels that do not contain River
Helmand Rōd >> Afghanistan	83
Indus >> China	1
Jordan >> Israel	2661
Orontes >> Lebanon	14
Rio Grande >> Mexico	1


OD 1 - Corresponding Water Basin Labels (most of these do have River) 
Helmand Rōd >> Afghanistan ( http://sws.geonames.org/7015667/ )	83
Indus River >> Pakistan ( http://sws.geonames.org/1115007/ )	1
Jordan River >> Israel ( http://sws.geonames.org/294624/ )	2661
Orontes River >> Lebanon ( http://sws.geonames.org/303365/ )	14
Rio Grande >> Mexico >> North America ( http://sws.geonames.org/3532104/ )	1

**Comment by morganem0** [comment]
@wickr I've finished Works-level QA for batches aa - an.
These look good to me.
I'm ready for ao and ap batches whenever they're ready. And then the MEW Works will be done!

The main issues I've seen in Works are (and maybe these are expected/non-issues?):
* Water basin labels missing 'River'
* Submission Date is different in OD2 vs. OD1. not sure if this matters

For the Water basin field, some OD2 labels are missing “River”, vs the values from OD1.
example:
In OD1, “Jordan River” ==> In OD2, “Jordan”
example Work metadata:
Water Basin: Jordan River >> Israel
vs.
Water Basin: Jordan >> Israel

more details about this in the "Water Basin labels missing 'River'" comment above.

**Comment by morganem0** [comment]
Works-level QA done for ao and ap batches, and I've updated all the batch-level tickets. 
All the minor errors noted in comments in this ticket have been resolved or ticketed for followup. 
After batch approvals complete I'll do a last facets check and then I think this is all set.

**Comment by carakey** [comment]
I have added the text from the additional / supplementary OD1 collection-level pages (Donor Bio pages) as a public page in the library wiki, and updated the OD2 collection metadata to point to the page as a Related URL.
